library/forms.inc

```diff
@@ -141,65 +119,57 @@ function getEncounters($pid, $dateStart = '', $dateEnd = '', $encounterRuleType
         $where .= "form_encounter.pid = ?";
         array_push($arraySqlBind, $pid);
     }
-
         $res = sqlStatement("SELECT distinct encounter FROM $from WHERE $where ORDER by date desc;", $arraySqlBind);
-
     for ($iter=0; $row=sqlFetchArray($res); $iter++) {
         $all[$iter] = $row;
     }
-
         return $all;
 }
-
 function getEncounterDateByEncounter($encounter)
 {
     global $attendant_type;
     $table = $attendant_type == 'pid' ? 'form_encounter' : 'form_groups_encounter';
     // $sql = "select date from forms where encounter='$encounter' order by date";
-    $sql = "select date from $table where encounter='$encounter' order by date";
-    return sqlQuery($sql);
+    $sql = "SELECT date from ? where encounter = ? order by date";
+    return sqlQuery($sql, array($table, $encounter));
 }
-
 function getProviderIdOfEncounter($encounter)
 {
         global $attendant_type;
         $table = $attendant_type == 'pid' ? 'form_encounter' : 'form_groups_encounter';
-        $sql = "select provider_id from $table where encounter=? order by date";
-        $res = sqlQuery($sql, array($encounter));
-        return $res['provider_id'];
+        $sql = "SELECT provider_id FROM ? WHERE encounter = ? ORDER BY date";
+        return sqlQuery($sql, array($encounter, $table))['provider_id'];
 }
-
 function getFormNameByFormdirAndFormid($formdir, $form_id)
-{
-        return sqlQuery("select form_name from forms where formdir = '$formdir' and form_id = '$form_id' and deleted = 0");
+{       
+        $sql = "SELECT form_name FROM forms WHERE formdir = ? AND form_id = ? AND deleted = 0";
+        return sqlQuery($sql, array($formdir, $form_id));
 }
-
 function getFormIdByFormdirAndFormid($formdir, $form_id)
-{
-    $result = sqlQuery("select id from forms where formdir = ? and form_id = ? and deleted = 0 ", array( $formdir, $form_id ));
+{   
+    $sql = "SELECT id FROM forms WHERE formdir = ? and form_id = ? and deleted = 0";
+    $result = sqlQuery($sql, array( $formdir, $form_id));
     return $result['id'];
 }
-
 function getFormNameByFormdir($formdir)
-{
-    return sqlQuery("select form_name from forms where formdir = '$formdir' and deleted = 0");
+{   
+    $sql = "SELECT form_name FROM forms WHERE formdir = ? and deleted = 0";
+    return sqlQuery($sql,array($formdir));
```

Comment:
in future, no need to break out the $sql string; for example, this could just be:
```
return sqlQuery("select form_name from forms where formdir = ? and deleted = 0", [$formdir]);
```